pkg/plugins/config/config.go

```diff
@@ -20,23 +18,6 @@ type Cfg struct {
 	DisablePlugins       []string
 	ForwardHostEnvVars   []string
 
-	// AWS Plugin Auth
```

Comment:
mmh, I am not sure what has been the criteria to remove some properties but not others? This Cfg struct is big but at least we can say that is a subset of the Grafana config for things that plugins care about. Now it's not clear to me what it means?

Reply:
The problem is we have multiple configs going on here _really_. `pkg/plugins/config` was originally meant for everything within `pkg/plugins`, IE all the core plugin management stuff, so that anything using plugin management could just provide what is necessary (the runner in this case). But the recent introduction of the request config stuff, it meant this config started to bloat with Grafana specific needs.

I think we could probably just stick with one in this case, so will have a look at reverting these changes!

Reply:
The main complication/annoying thing here is that both still need to rely on `setting.Provider`, which is based off of the Grafana config. It feels a bit weird to mix the plugins specific config and this together, rather than just the Grafana config.

Tests will need to be something like:

```go
cfg := setting.NewCfg()
s := NewRequestConfigProvider(&config.Cfg{}, setting.ProvideProvider(cfg), featuremgmt.WithFeatures())
```

and

```go
cfg := setting.NewCfg()
s := NewPluginEnvVarsProvider(&config.Cfg{}, setting.ProvideProvider(cfg), featuremgmt.WithFeatures(), licensing)
```

Reply:
I just took some time to split out a "plugin management" config and a "plugin instance config". LMK if you still prefer a single one though. I figure the sooner we get used to smaller more distinct config structs the better. This feels right, but happy to iterate on it further!

Reply:
sounds good :+1: what do you think about changing the name of this struct to make it more clear?